src/main/java/com/xebia/fs101/zohoreplica/controller/UserController.java

```diff
@@ -0,0 +1,36 @@
+package com.xebia.fs101.zohoreplica.controller;
+
+
+import com.xebia.fs101.zohoreplica.api.constant.ApplicationConstant;
+import com.xebia.fs101.zohoreplica.api.request.UserRequest;
+import com.xebia.fs101.zohoreplica.api.response.ZohoReplicaResponse;
+import com.xebia.fs101.zohoreplica.entity.User;
+import com.xebia.fs101.zohoreplica.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+@RestController
+public class UserController {
+
+
+    @Autowired
+    private UserService userService;
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> signup(@Valid @RequestBody UserRequest userRequest) {
+
+        User savedUser = userService.save(userRequest);
```

Comment:
userRequest should not go to the service layer. Convert it to the entity User and the save.
You can merge then.